OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -39,23 +42,59 @@ public FindResources(Actor self)
 			claimLayer = self.World.WorldActor.Trait<ResourceClaimLayer>();
 			pathFinder = self.World.WorldActor.Trait<IPathFinder>();
 			domainIndex = self.World.WorldActor.Trait<DomainIndex>();
+			this.deliverActor = deliverActor;
 		}
 
-		public FindResources(Actor self, CPos avoidCell)
-			: this(self)
+		public FindResources(Actor self, CPos orderLocation)
+			: this(self, null)
 		{
-			this.avoidCell = avoidCell;
+			this.orderLocation = orderLocation;
+		}
+
+		protected override void OnFirstRun(Actor self)
+		{
+			// If an explicit order is given, direct the harvester to the ordered location instead of the previous
+			// harvested cell for the initial search.
+			if (orderLocation != null)
+				harv.LastHarvestedCell = orderLocation;
 		}
 
 		public override Activity Tick(Actor self)
 		{
-			if (IsCanceling)
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+				if (ChildActivity != null)
+					return this;
+
+				// Interrupt automated harvesting after first harvest activity, so explicit orders can be queued.
```

Comment:
IMO "harvest activity" isn't as clear as it could be.

Assuming we change `QueueRuns` to `QueueFullLoad` (see comment above), can we change this to
```
// Move to the next activity after clearing the first cell
```
?

Reply:
Done.